baseplate/frameworks/queue_consumer/kombu.py

```diff
@@ -0,0 +1,192 @@
+import logging
+import queue
+import socket
+
+from typing import Callable
+from typing import Optional
+from typing import Sequence
+from typing import TYPE_CHECKING
+
+import kombu
+
+from gevent.server import StreamServer
+from kombu.mixins import ConsumerMixin
+from kombu.transport.virtual import Channel
+from pyramid.config import Configurator
+
+from baseplate import Baseplate
+from baseplate import RequestContext
+from baseplate.clients.kombu import KombuSerializer
+from baseplate.frameworks.pyramid import BaseplateConfigurator
+from baseplate.server.queue_consumer import MessageHandler
+from baseplate.server.queue_consumer import PumpWorker
+from baseplate.server.queue_consumer import QueueConsumerFactory
+from baseplate.server.wsgi import make_server as make_wsgi_server
+
+
+if TYPE_CHECKING:
+    WorkQueue = queue.Queue[kombu.Message]  # pylint: disable=unsubscriptable-object
+else:
+    WorkQueue = queue.Queue
+
+
+logger = logging.getLogger(__name__)
+
+
+Handler = Callable[[RequestContext, kombu.Message], None]
+HealthCheck = Callable[[RequestContext], bool]
+
+
+class AbortError(Exception):
+    """An error that signals that the queue process should exit.
+
+    Raising an Exception that is a subclass of AbortError will cause the
+    KombuMessageHandler to re-raise the exception rather than swallowing it
+    which will cause the handler thread/process to stop.  This, in turn, will
+    gracefully shut down the QueueConsumerServer currently running.
+
+    Exceptions of this nature should be reserved for errors that are due to
+    problems with the environment rather than the message itself.  For example,
+    a node that cannot get its AWS credentials.
+    """
+
+
+class KombuConsumerWorker(ConsumerMixin, PumpWorker):  # pylint: disable=abstract-method
```

Comment:
why disable abstract-method?

Reply:
`run` is defined by `ConsumerMixin` but the linter doesn't seem to be aware of that, I'll try running it again though just to double check.

Reply:
if you swap the order of the base classes does that change it?

Reply:
The order actually _is_ already swapped from when I first ran this, I actually update the order because it wasn't working.  So pylint caught my problem and I told it to go away. 🤣 

I confirmed that I don't need this when the code actually works.